lib/pure/httpclient.nim

```diff
@@ -1166,9 +1167,15 @@ proc request*(client: HttpClient | AsyncHttpClient, url: string,
   var lastURL = url
   for i in 1..client.maxRedirects:
     if result.status.redirection():
+      # we get all sorts of troubles if we don't close the connection
+      # before following a redirect. this is inefficient but i didn't
+      # find a better solution.
+      client.close()
       let redirectTo = getNewLocation(lastURL, result.headers)
-      result = await client.request(redirectTo, httpMethod, body, headers)
+      result = await client.requestAux(redirectTo, httpMethod, body, headers)
```

Comment:
This makes sense to me, it should have been changed in a previous PR that moved the redirection logic into ``request``.